kedro-datasets/test_requirements.txt

```diff
@@ -24,13 +24,15 @@ lxml~=4.6
 matplotlib>=3.0.3, <3.4; python_version < '3.10' # 3.4.0 breaks holoviews
 matplotlib>=3.5, <3.6; python_version == '3.10'
 memory_profiler>=0.50.0, <1.0
+mlflow~=1.30.0; python_version=='3.7'
+mlflow~=2.2.1; python_version>='3.8'
 moto==1.3.7; python_version < '3.10'
 moto==3.0.4; python_version == '3.10'
 networkx~=2.4
 opencv-python~=4.5.5.64
 openpyxl>=3.0.3, <4.0
 pandas-gbq>=0.12.0, <0.18.0
-pandas>=1.3  # 1.3 for read_xml/to_xml
+pandas>=1.3, <2  # 1.3 for read_xml/to_xml, <2 for compatibility with Spark < 3.4
```

Comment:
Why do pandas dependency matter here? I expect Pyspark has it own pandas dependency pin already (if needed)

Reply:
Pandas >= 2 removed a method that that Spark < 3.4.0 relies on. This was fixed in Spark 3.4.0. With Spark < 3.4.0 and Pandas >= 2, new tests fail.

There are two options for getting tests to pass here:

1. require PySpark >= 3.4.0, or
2. require Pandas < 2.0

I went for option 2, as pinning PySpark ~= 3.4.0 (and therefore Spark 3.4.0) causes a dependency issue with `delta-spark` and causes all new tests to fail (plus Spark tests to fail at the import statements), whereas requiring Pandas < 2.0 caused no other problems with the test suite. Since there is some pressure to get this released and getting our tests to work with Spark 3.4.0 will take some work, I opted to limit Pandas instead.

I would keep it like this, then open an issue to upgrade our Spark to 3.4.0 asap. I can tackle this issue after the Databricks work is delivered.